static/app/components/searchQueryBuilder/combobox.tsx

```diff
@@ -24,7 +24,8 @@ type SearchQueryBuilderComboboxProps = {
   inputLabel: string;
   inputValue: string;
   items: SelectOptionWithKey<string>[];
-  onCustomValueSelected: (value: string) => void;
+  onCustomValueBlurred: (value: string) => void;
+  onCustomValueCommitted: (value: string) => void;
```

Comment:
Instead of a single `onCustomValueSelected`, I broke it into two callbacks. One which fires when the value was blurred, and the other when it was specifically committed with enter.